### PR TITLE
trunk placer and tree generation mappings

### PR DESCRIPTION
**Pull request by liach**
Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

File: mappings/net/minecraft/world/gen/trunk/TrunkPlacer.mapping [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		ARG 2 random
+		ARG 3 trunkHeight
+		ARG 4 pos
+		ARG 5 foliageRadius
+		ARG 6 logs
+		ARG 7 box
+		ARG 8 config
+	METHOD method_26992 serialize (Lcom/mojang/datafixers/types/DynamicOps;)Ljava/lang/Object;
+		ARG 1 ops
+	METHOD method_26993 getHeight (Ljava/util/Random;Lnet/minecraft/class_4640;)I
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/trunk/TrunkPlacer.mapping:25]: I assume this method returns the height after the random was applied, so maybe `getFinalHeight` would be better?

Comment (liach) [reply mappings/net/minecraft/world/gen/trunk/TrunkPlacer.mapping:25]: the height field should probably be renamed to base height than editing this method

**Review by Runemoro — APPROVED** [review]

File: mappings/net/minecraft/world/gen/trunk/TrunkPlacer.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/world/gen/trunk/TrunkPlacer.mapping:25]: However the `height` and `height_random` are fields from mojang, I can still rename `height` to `baseHeight`, `offset` to `baseOffset` if you'd prefer. @Runemoro how's your opinion?

Comment (Runemoro) [reply mappings/net/minecraft/world/gen/trunk/TrunkPlacer.mapping:25]: I think it's clear `getHeight` is the final height. Any other height would be named something like `getBaseHeight`.

Comment (liach) [reply mappings/net/minecraft/world/gen/trunk/TrunkPlacer.mapping:25]: There is a field in this class called `height` (as seen in data structure), but the field is effectively a base height. Should that field be renamed from `height` to `baseHeight`

Comment (Runemoro) [reply mappings/net/minecraft/world/gen/trunk/TrunkPlacer.mapping:25]: Oh. If it matches the Mojang name then I think it's ok.

File: mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping [hunk 2]
```diff
@@ -26,9 +47,14 @@ CLASS net/minecraft/class_4647 net/minecraft/world/gen/foliage/FoliagePlacer
 		ARG 5 leaves
 	METHOD method_23451 isInvalidForLeaves (Ljava/util/Random;IIIII)Z
 		ARG 2 baseHeight
-		ARG 3 x
-		ARG 4 y
-		ARG 5 z
+		ARG 3 dx
+		ARG 4 dy
+		ARG 5 dz
```
Comment (jaskarth) [inline mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping:52]: why was this changed?

File: mappings/net/minecraft/world/gen/trunk/ForkingTrunkPlacer.mapping [hunk 3]
```diff
@@ -0,0 +1,7 @@
+CLASS net/minecraft/class_5139 net/minecraft/world/gen/trunk/ForkingTrunkPlacer
```
Comment (jaskarth) [inline mappings/net/minecraft/world/gen/trunk/ForkingTrunkPlacer.mapping:1]: wouldn't `BranchedTrunkPlacer` be a better name? since we already refer to certain things as `Branched`

File: mappings/net/minecraft/world/gen/trunk/StraightTrunkPlacer.mapping [hunk 4]
```diff
@@ -0,0 +1,7 @@
+CLASS net/minecraft/class_5140 net/minecraft/world/gen/trunk/StraightTrunkPlacer
+	METHOD <init> (III)V
+		ARG 1 baseHeight
+		ARG 2 firstRandomHeight
+		ARG 3 secondRandomHeight
```
Comment (jaskarth) [inline mappings/net/minecraft/world/gen/trunk/StraightTrunkPlacer.mapping:5]: weren't these referred to as `heightRandA` and `heightRandB` before?

File: mappings/net/minecraft/world/gen/trunk/ForkingTrunkPlacer.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/world/gen/trunk/ForkingTrunkPlacer.mapping:1]: This is from mojang's registration id

File: mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping:52]: because this is not xyz but relative xyz from the tree root

File: mappings/net/minecraft/world/gen/trunk/StraightTrunkPlacer.mapping (hunk 4 above)
Comment (liach) [reply mappings/net/minecraft/world/gen/trunk/StraightTrunkPlacer.mapping:5]: heightRandA and heightRandB are not descriptive and doesn't fit with yarn guideline

File: mappings/net/minecraft/world/gen/trunk/ForkingTrunkPlacer.mapping (hunk 3 above)
Comment (jaskarth) [reply mappings/net/minecraft/world/gen/trunk/ForkingTrunkPlacer.mapping:1]: god damn it

File: mappings/net/minecraft/world/gen/trunk/StraightTrunkPlacer.mapping (hunk 4 above)
Comment (jaskarth) [reply mappings/net/minecraft/world/gen/trunk/StraightTrunkPlacer.mapping:5]: alright

File: mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping (hunk 2 above)
Comment (jaskarth) [reply mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping:52]: ok

**Review by jaskarth — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]
